tests/phpunit/tests/media/shortcodes/test-wpml-page-builders-media-shortcodes.php

```diff
@@ -141,6 +141,181 @@ public function it_should_translate() {
+		[et_pb_video_slider_item _builder_version="3.12" image_src="' . $original_url_2 . '" background_layout="dark" src_webm="http://wpml.local/video2.webm" /]
+		[et_pb_video_slider_item _builder_version="3.12" image_src="' . $original_url_1 . '" background_layout="dark" src_webm="http://wpml.local/video1.webm" /]
+	[/et_pb_video_slider]
+	[et_pb_video image_src="' . $original_url_1 . '" some_image_src="Should not be translated" background_image="' . $original_url_2 . '" /]
+	[shortcode_with_single_quotes gallery_ids=\'' . $original_id_1 . '\' /]
+	[shortcode_with_url_content foo="bar"]' . $original_url_1 . '[/shortcode_with_url_content]
+	[shortcode_with_url_content]' . $original_url_2 . '[/shortcode_with_url_content]
+[/et_pb_column][/et_pb_row][/et_pb_section]
+';
+
+		$config = array(
+			array(
+				'tag'        => array( 'name' => 'et_pb_cta' ),
+				'attributes' => array(
+					'background_image' => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_URL ),
+				),
+			),
+			array(
+				'tag'        => array( 'name' => 'et_pb_audio' ),
+				'attributes' => array(
+					'background_image' => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_URL ),
+				),
+			),
+			array(
+				'tag'        => array( 'name' => 'et_pb_video_slider_item' ),
+				'attributes' => array(
+					'image_src' => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_URL ),
+				),
+			),
+			array(
+				'tag'        => array( 'name' => 'et_pb_video' ),
+				'attributes' => array(
+					'image_src'        => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_URL ),
+					'background_image' => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_URL ),
+				),
+			),
+			array(
+				'tag'        => array( 'name' => 'et_pb_gallery' ),
+				'attributes' => array(
+					'gallery_ids' => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_IDS ),
+				),
+			),
+			array(
+				'tag'        => array( 'name' => 'shortcode_with_single_quotes' ),
+				'attributes' => array(
+					'gallery_ids' => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_IDS ),
+				),
+			),
+			array(
+				'tag'     => array( 'name' => 'shortcode_with_url_content' ),
+				'content' => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_URL ),
+			),
+			// Missing tag, should not alter content
+			array(
+				'attributes' => array(
+					'background_image' => array( 'type' => WPML_Page_Builders_Media_Shortcodes::TYPE_URL ),
+				),
+			),
+			// No attributes, should not alter content
+			array(
+				'tag' => array( 'name' => 'et_pb_gallery' ),
+			),
+		);
+
+		$media_translate = $this->get_media_translate();
+
+		$subject = $this->get_subject( $media_translate, $config );
+
+		$this->assertTrue( $subject->has_media_shortcode( $content ) );
+	}
+
+	/**
+	 * @test
+	 * @group wpmlpb-166
+	 */
+	public function it_should_return_false_when_content_has_NO_media_shortcodes() {
+		$content = '[sr]';
+
+		$config = array(
```

Comment:
Same for this test, you can have only two shortcodes (not matching).

Reply:
Done